server/game/cards/events/01/olennascunning.js

```diff
@@ -1,35 +1,31 @@
 const DrawCard = require('../../../drawcard.js');
 
 class OlennasCunning extends DrawCard {
-    canPlay(player, card) {
-        if(this !== card || !this.game.currentChallenge || (this.game.currentChallenge.challengeType !== 'intrigue' && this.game.currentChallenge.challengeType !== 'power')) {
-            return false;
-        }
-
-        if(this.game.currentChallenge.winner !== this.controller) {
-            return false;
-        }
-
-        return super.canPlay(player, card);
-    }
-
-    play() {
-        var buttons = [
-            { text: 'Character', method: 'typeSelected', arg: 'character' },
-            { text: 'Location', method: 'typeSelected', arg: 'location' },
-            { text: 'Attachment', method: 'typeSelected', arg: 'attachment' },
-            { text: 'Event', method: 'typeSelected', arg: 'event' }
-        ];
-
-        this.game.promptWithMenu(this.game.currentChallenge.loser, this, {
-            activePrompt: {
-                menuTitle: 'Select a card type',
-                buttons: buttons
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    ['intrigue', 'power'].includes(challenge.challengeType) &&
```

Comment:
I generally prefer this pattern to doing an explicit or since it's much more compact, but understand the readability pushback. I've been toying with adding an isMatch function to challenge so we can do things like this instead:
```javascript
challenge.isMatch({
    challengeType: ['intrigue', 'power'],
    winner: this.controller
})
```
which would be both more readable and more compact than what we're currently doing. But there are subtleties that have held me back from doing it, and it seems like by the time I get around to it we'll have pretty much all cards implemented. :woman_shrugging: